### PR TITLE
fix(EMS-3560): update az cli to v2

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Update Azure CLI GHA plugin to `v2`

## Resolution :heavy_check_mark:
* Updated `azure/cli` to `v2`.
* Refactored `lint.yml` and `sca.yml`.
* Added two distinct DNS domains names for public Azure DNS Zones.

## Miscellaneous :heavy_plus_sign:
* Minor lint fixes
* Updated `szenius/set-timezone` plugin to `v2`

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -125,7 +125,64 @@ jobs:
...
+      - name: Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ vars.NODE_VERSION }}
+
+      - name: Dependencies
+        working-directory: ./
+        run: npm run ci:all
+
+      # TODO: Enable post TSC error resolution
```
Comment (ttbarnes) [inline .github/workflows/lint.yml:157]: Do we need a ticket for this?

Comment (abhi-markan) [reply .github/workflows/lint.yml:157]: Yes we do, would you like me to add ticket numbers to all TODO?

File: .github/workflows/deployment.yml [hunk 2]
```diff
@@ -73,21 +73,21 @@ jobs:
           creds: ${{ secrets.AZURE_CREDENTIALS }}
 
       - name: Defaults ✨
-        uses: azure/cli@v2.0.0
+        uses: azure/cli@v2
```
Comment (Zainzzkk) [inline .github/workflows/deployment.yml:76]: could this not run the risk of updating to the latest version which could have breaking changes? my mysql?

Comment (abhi-markan) [reply .github/workflows/deployment.yml:76]: It shouldn't do as this is just Azure CLI version update, if there is a breaking change then pipelines will pick it up.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (abhi-markan) [reply .github/workflows/lint.yml:157]: Have added ticket numbers to `TODO`s. Thanks

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**